server/odc-service/src/main/java/com/oceanbase/odc/service/iam/auth/BaseAuthorizer.java

```diff
@@ -43,6 +44,27 @@ public void checkPermission(Principal principal, Collection<Permission> permissi
         }
     }
 
+    @Override
+    public boolean isPermitted(Principal principal, Collection<Permission> permissions, SecurityContext context) {
+        User odcUser = (User) principal;
+        if (Objects.isNull(odcUser.getId())) {
+            return false;
+        }
+        for (Permission permission : permissions) {
+            boolean accessDenied = true;
+            for (Permission resourcePermission : listPermittedPermissions(principal)) {
```

Comment:
will listPermittedPermissions access remote storage? 
if so can reuse the result first queried?
Is this logic want's to describe a rule that any resource permission not permitted by  listPermittedPermissions  will return false?

Reply:
> will listPermittedPermissions access remote storage?
> if so can reuse the result first queried?

Yes, it will access to the metadb. And you are right, if should be reused, and I updated it.

> Is this logic want's to describe a rule that any resource permission not permitted by listPermittedPermissions will return false?

Yes, your understanding is right.